src/chains/ethereum/ethereum/tests/executables.test.ts

```diff
@@ -0,0 +1,138 @@
+          bootstrapNodes: []
+        },
+        "london"
+      );
+
+      const [from, from2] = wallet.addresses;
+      const secretKey = wallet.unlockedAccounts.get(from).toBuffer();
+      const secretKey2 = wallet.unlockedAccounts.get(from2).toBuffer();
+      const rpcTransaction: Transaction = {
+        from,
+        type: "0x2",
+        maxFeePerGas: "0xffffffff",
+        maxPriorityFeePerGas: "0xff",
+        gasLimit: "0xffff"
+      };
+
+      const { inProgress, pending } = executables;
+      // set up the heap to store all pending transactions for the from address
+      const heapForFrom = new Heap(byNonce);
+      pending.set(from, heapForFrom);
+      // we'll add 10 transactions to executables
+      for (let i = 0; i < 10; i++) {
+        const transaction = TransactionFactory.fromRpc(
+          { ...rpcTransaction, nonce: Quantity.toString(i) },
+          common
+        );
+        // make them all "legit"
+        transaction.signAndHash(secretKey);
+        if (i < 5) {
+          // add the first 5 to the in progress queue
+          transaction.locked = true;
+          inProgress.add(transaction);
+        } else {
+          // the rest will go in the pending queue
+          if (i === 7) {
+            // but one will be "locked" (starting to be mined)
+            transaction.locked = true;
+          }
+          heapForFrom.push(transaction);
+        }
+      }
+      // we'll also set up another heap for our second origin
+      const heapForFrom2 = new Heap(byNonce);
+      pending.set(from2, heapForFrom2);
+      const transaction = TransactionFactory.fromRpc(
+        { ...rpcTransaction, nonce: "0x0", from: from2 },
+        common
+      );
+      transaction.signAndHash(secretKey2);
+      heapForFrom2.push(transaction);
+
+      // finally, clone and reset
+      clone = executables.cloneAndReset();
+      // and make some data easier to compare in tests
+      const { pending: clonedPending } = clone;
+      clonedTransactions = getAllTransactionsFromPending(clonedPending);
+      originalTransactions = [
+        ...Array.from(inProgress),
+        ...getAllTransactionsFromPending(pending)
+      ];
+    });
+
+    it("creates a clone that does not have any `inProgress` transactions ", async () => {
+      assert.strictEqual(
+        clone.inProgress.size,
+        0,
+        "Expected cloned `inProgress` pool to have no transactions."
+      );
+    });
+
+    it("creates a clone that has all of the same transactions as the original", async () => {
+      assert.strictEqual(
+        clonedTransactions.length,
+        originalTransactions.length
+      );
+      for (const originalTransaction of originalTransactions) {
+        const clonedTransaction = clonedTransactions.find(transaction => {
+          return transaction.hash === originalTransaction.hash;
+        });
+        // there is a transaction with the same hash
+        assert(clonedTransaction);
```

Comment:
```suggestion
        assert(clonedTransaction !== undefined);
```
I'd rather this were explicit, purely for the sake of readability.

Reply:
Can we also assert that, although the hashes match, it is, infact, a clone and not the same instance?

Reply:
We can't assert (without a good bit more code) that they are exact clones (because they aren't, some of the data doesn't get cloned). But I did make an assertion that they aren't the same instance.